test/framework/easyconfig.py

```diff
@@ -1876,6 +1892,244 @@ def test_dump_comments(self):
+            },
+            'sanity_check_paths': {
+                "    'dirs': [],": ['# no dirs!'],
+                "    'files': ['bin/foo', 'lib/libfoo.a'],": ['# a bunch of files'],
+            },
+            'source_urls': {
+                "    'https://example.com',": ['# first possible source URL'],
+            },
+        })
+        self.assertEqual(ec.parser._formatter.comments['iterinline'], {
+            'dependencies': {
+                "    ('foo', '1.2.3'),": '  # and an inline comment too',
+                ']': "  # inline comment after closing dependencies",
+            },
+            'source_urls': {
+                "    'https://anotherexample.com',": '  # fallback URL',
+            },
+        })
+        self.assertEqual(ec.parser._formatter.comments['tail'], [
+            '#',
+            '# trailing comment',
+            '',
+            '# with an empty line in between',
+            '# DONE!',
+        ])
+
+        ec.dump(testec)
+        ectxt = read_file(testec)
+
+        # reparsing the dumped easyconfig file should work
+        EasyConfig(testec)
+
+        check_easyconfigs_style([testec])
+
+        self.assertTrue(ectxt.startswith('\n'.join([
+            '# this is a header',
+            '#',
+            '# which may include empty comment lines',
+            '# weirdly indented lines',
+            '',
+            '# or flat out empty lines',
+            '',
+        ])))
+
+        patterns = [
+            # inline comments
+            r"^homepage = .*  # may not be actual homepage\n",
+            r"^name = .*  # just a test",
+            r"^    \('foo', '1\.2\.3'\),  # and an inline comment too",
+            r"^    'https://anotherexample\.com',  # fallback URL",
+            # comments above parameter definition
+            '\n'.join([
+                r'',
+                r"# this is a multiline comment above dependencies",
+                r"# I said multiline",
+                r"# multi > 3",
+                r"dependencies = ",
+            ]),
+            '\n'.join([
+                r'',
+                r"# how about comments above and in a dict value\?",
+                r"sanity_check_paths = ",
+            ]),
+            '\n'.join([
+                r'',
+                r"# how about # a comment with # multple additional hashes",
+                r"source_urls = ",
+            ]),
+            '\n'.join([
+                r'',
+                r"# after toolchain, before sources comment",
+                r'',
+                r"# this comment contains another #, uh-oh...",
+                r"sources = ",
+            ]),
+            '\n'.join([
+                r'',
+                r"# the version doesn't matter much here",
+                r"version = ",
+            ]),
+            '\n'.join([
+                r'',
+                r"    # trailing comments in dependencies",
+                r"    # a bit weird, but it happens",
+                r"\]",
```

Comment:
Might as well add the inline comment to the test for the closing ].

Reply:
done in f7c14da